torch/_prims/__init__.py

```diff
@@ -2416,6 +2422,73 @@ def _full_like_aten(
     doc=_full_like_doc,
 )
 
+
+#
+# Linear algebra (linalg) prims
+#
+
+
+def _svd_meta(
+    A: TensorLikeType, *, full_matrices: bool
+) -> Tuple[TensorLikeType, TensorLikeType, TensorLikeType]:
+    A_shape = A.shape
+    check(
+        len(A_shape) >= 2,
+        lambda: f"linalg.svd: input should have at least 2 dimensions, but has {A.ndim} dimensions instead",
+    )
+    batch = A_shape[:-2]
+    m, n = A_shape[-2:]
+    k = min(m, n)
+
+    def contig_strides(shape, *, row_major):
+        if row_major:
+            return utils.make_contiguous_strides_for(shape)
+        else:  # row major
+            return utils.make_batched_column_major_strides_for(shape)
+
+    shape_U = batch + (m, m if full_matrices else k)
+    strides_U = contig_strides(shape_U, row_major=False)
+    U = TensorMeta(shape=shape_U, strides=strides_U, dtype=A.dtype, device=A.device)
+
+    shape_S = batch + (k,)
+    strides_S = contig_strides(shape_S, row_major=True)
+    S = TensorMeta(
+        shape=shape_S,
+        strides=strides_S,
+        dtype=utils.corresponding_real_dtype(A.dtype),
+        device=A.device,
+    )
+
+    shape_Vh = batch + (n if full_matrices else k, n)
+    # The CPU backend returns V, but the cuSolver backend returns V^H
+    # TODO The MAGMA backend returns V, so this is wrong if used with the MAGMA backend
+    is_cuda = A.device.type == "cuda"
+    strides_Vh = contig_strides(shape_Vh, row_major=is_cuda)
+    Vh = TensorMeta(shape=shape_Vh, strides=strides_Vh, dtype=A.dtype, device=A.device)
+    return U, S, Vh
+
+
+def _svd_aten(
+    A: TensorLikeType, *, full_matrices: bool
+) -> Tuple[Tensor, Tensor, Tensor]:
+    return torch.linalg.svd(A, full_matrices=full_matrices)
+
+
+_svd_doc = """
+    Returns the SVD of a matrix or batch of matrices.
+
+    The `full_matrices` flag controls whether the full or reduced SVD decomposition is returned.
```

Comment:
Future question: should the full and reduced SVD be different prims?

Reply:
Probably not. If we're going in the direction of ending up exposing something that smells like the LAPCK / cuSolver API and then write the dispatch in Python, we probably want them to be the same operation